src/lib/Scenes/MyProfile/MyProfilePushNotifications.tsx

```diff
@@ -0,0 +1,315 @@
+
+interface SwitchMenuProps {
+  onChange: (value: boolean) => void
+  value: boolean
+  title: string
+  description: string
+}
+
+export type PushNotificationPermissions =
+  | "receiveLotOpeningSoonNotification"
+  | "receiveNewSalesNotification"
+  | "receiveNewWorksNotification"
+  | "receiveOutbidNotification"
+  | "receivePromotionNotification"
+  | "receivePurchaseNotification"
+  | "receiveSaleOpeningClosingNotification"
+
+export const SwitchMenu = ({ onChange, value, title, description }: SwitchMenuProps) => (
+  <Flex flexDirection="row" alignItems="flex-start" flexShrink={0} my={1}>
+    <Flex style={{ width: "80%" }}>
+      <Sans size="4t" color="black100">
+        {title}
+      </Sans>
+      <Sans size="3t" color={colors["gray-semibold"]} py={0.5}>
+        {description}
+      </Sans>
+    </Flex>
+    <Flex style={{ width: "20%" }} alignItems="flex-end">
+      <Switch
+        trackColor={{ false: color("black10"), true: color("black100") }}
+        onValueChange={onChange}
+        value={value}
+      />
+    </Flex>
+  </Flex>
+)
+
+export const MyProfilePushNotifications: React.FC<{
+  me: MyProfilePushNotifications_me
+  relay: RelayRefetchProp
+}> = ({ me, relay }) => {
+  // TODO: This will be replaced with a custom hook to get the permission
+  const [hasPushNotificationsEnabled] = useState<boolean>(true)
+  const [isRefreshing, setIsRefreshing] = useState<boolean>(false)
+
+  const onForeground = useCallback(async () => {
+    try {
+      // Enable all push notification options if the user enabled push notifications
+      if (hasPushNotificationsEnabled) {
+        await updateMyProfilePushNotifications({
+          receiveLotOpeningSoonNotification: true,
+          receiveNewSalesNotification: true,
+          receiveNewWorksNotification: true,
+          receiveOutbidNotification: true,
+          receivePromotionNotification: true,
+          receivePurchaseNotification: true,
+          receiveSaleOpeningClosingNotification: true,
+        })
+      }
+    } catch (error) {
+      Alert.alert(typeof error === "string" ? error : "Something went wrong.")
+    }
+  }, [hasPushNotificationsEnabled])
+
+  useAppState({ onForeground })
+
+  const onRefresh = useCallback(() => {
+    setIsRefreshing(true)
+    relay.refetch(() => {
+      setIsRefreshing(false)
+    })
+  }, [])
+
+  const updateNotificationPermission = async (notificationType: PushNotificationPermissions, value: boolean) => {
+    try {
+      await updateMyProfilePushNotifications({ ...me, [notificationType]: value })
+    } catch (error) {
+      Alert.alert(typeof error === "string" ? error : "Something went wrong.")
+    }
+  }
+
+  // Render turn on notifications banner
+  const renderAllowPushNotificationsBanner = () => {
+    return (
+      <>
+        <Flex py={3} px={2} backgroundColor={colors["gray-light"]} alignItems="center">
```

Comment:
this should probably be "black5"
```suggestion
        <Flex py={3} px={2} backgroundColor="black5" alignItems="center">
```